phac-nml/irida-next
# Select sample table update

## What does this PR do and why?
Modified the checkbox within the samples table header to select the rows on the current page.
Added new `Select All` and `Deselect All` buttons that selects/deselects all table rows.

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/82f6bff1-e060-4b19-ba97-a2f964ef2a12)
...
![image](https://github.com/user-attachments/assets/3e84a98f-fc3d-4602-a9fd-f2c0aa7f26d3)

## How to set up and validate locally
1. Go to a group or project samples page.
2. Click the `Select All` button.
3. Verify the all the samples are selected and the table footer has been updated.
4. Click the `Deselect All` button.
5. Verify the all the samples are selected and the table footer has been updated.
6. Check the checkbox within the table header.
7. Verify the all the samples on the page are selected and the table footer has been updated.
8. Uncheck the checkbox within the table header.
9. Verify the all the samples on the page are unselected and the table footer has been updated.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
